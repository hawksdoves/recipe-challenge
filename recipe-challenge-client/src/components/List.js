import React from 'react';
import { Link } from 'react-router-dom';
import { Card } from 'semantic-ui-react';

import Recipe from './Recipe';
import Favourite from './Favourite';

function List(props) {
    const { recipes, favourites, ...rest } = props;

    return recipes.length > 0 ?
        (<Card.Group >
            { recipes.map(recipe => {
                return (  
                    <div key={recipe.id} className='recipe' >
                        <Link to={`/recipe/${recipe.id}`} key={`Link-${recipe.id}`} className='card'>
                            <Recipe recipe={recipe} />
                        </Link>
                        <Favourite favourite={favourites.includes(recipe.id)} {...rest} recipeId={recipe.id} />
                    </div>
                )
            })}
        </Card.Group> ) :
        <div>Sorry, nothing matched your filter term</div>
};

export default List;
