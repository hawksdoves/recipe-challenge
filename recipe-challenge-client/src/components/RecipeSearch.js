import React from 'react';

import List from './List';

class RecipeSearch extends React.Component {

    constructor(props) {
        super(props)
        this.state = { recipeResults: props.recipes }
        this.searchRecipes = this.searchRecipes.bind(this)
    }

    searchRecipes(event) {
        const updatedResults = this.props.recipes.filter(recipe => {
            return recipe.name.toLowerCase().search(
                event.target.value.toLowerCase()) !== -1;
            });
        this.setState({ recipeResults: updatedResults});
    }

    render(props) {
        return (
            <div>
                <input type="text" placeholder="Recipe Search" onChange={this.searchRecipes} />
                { this.state.recipeResults.length > 0 ?
                    <List recipes={this.state.recipeResults} /> :
                    <div>Sorry, nothing matched your filter term</div>
                }
            </div>
        )
    }
}

export default RecipeSearch;
