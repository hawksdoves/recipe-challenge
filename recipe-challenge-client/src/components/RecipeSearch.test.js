import React from 'react';
import { shallow } from 'enzyme';

import RecipeSearch from './RecipeSearch';
import List from './List';

describe('RecipeSearch', () => {
    const recipes = [
        {
            id: '1',
            name: 'steak',
            ingredients: [{
                name: 'steak',
                quantity: '3'
            }]
        },
        {
            id: '2',
            name: 'chicken',
            ingredients: [{
                name: 'chicken',
                quantity: '2'
            }]
        }
    ]

    it('renders List', () => {
        const wrapper = shallow(<RecipeSearch recipes={recipes}/>)
        expect(wrapper).toMatchSnapshot()
    })

    it('only displays recipes that match search', () => {
        const wrapper = shallow(<RecipeSearch recipes={recipes}/>)
        wrapper.find('input').simulate('change', { target: { value: 'steak' } })
        expect(wrapper.find(List).length).toEqual(1)
        expect(wrapper).toMatchSnapshot()
    })

    it('renders error message if search had no matches', () => {
        const wrapper = shallow(<RecipeSearch recipes={recipes}/>)
        wrapper.find('input').simulate('change', { target: { value: 'fish' } })
        expect(wrapper.find(List).length).toEqual(0)
        expect(wrapper).toMatchSnapshot()
    })
});
