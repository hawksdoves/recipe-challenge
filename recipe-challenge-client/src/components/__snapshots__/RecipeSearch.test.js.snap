// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeSearch only displays recipes that match search 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <input
        onChange={[Function]}
        placeholder="Recipe Search"
        type="text"
    />
    <List
        recipes={
            Array [
                Object {
                  "id": "1",
                  "ingredients": Array [
                    Object {
                      "name": "steak",
                      "quantity": "3",
                    },
                  ],
                  "name": "steak",
                },
              ]
        }
    />
</div>,
  "nodes": Array [
    <div>
      <input
            onChange={[Function]}
            placeholder="Recipe Search"
            type="text"
      />
      <List
            recipes={
                  Array [
                        Object {
                          "id": "1",
                          "ingredients": Array [
                            Object {
                              "name": "steak",
                              "quantity": "3",
                            },
                          ],
                          "name": "steak",
                        },
                      ]
            }
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <RecipeSearch
        recipes={
                Array [
                        Object {
                          "id": "1",
                          "ingredients": Array [
                            Object {
                              "name": "steak",
                              "quantity": "3",
                            },
                          ],
                          "name": "steak",
                        },
                        Object {
                          "id": "2",
                          "ingredients": Array [
                            Object {
                              "name": "chicken",
                              "quantity": "2",
                            },
                          ],
                          "name": "chicken",
                        },
                      ]
        }
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": RecipeSearch {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "recipes": Array [
            Object {
              "id": "1",
              "ingredients": Array [
                Object {
                  "name": "steak",
                  "quantity": "3",
                },
              ],
              "name": "steak",
            },
            Object {
              "id": "2",
              "ingredients": Array [
                Object {
                  "name": "chicken",
                  "quantity": "2",
                },
              ],
              "name": "chicken",
            },
          ],
        },
        "refs": Object {},
        "searchRecipes": [Function],
        "state": Object {
          "recipeResults": Array [
            Object {
              "id": "1",
              "ingredients": Array [
                Object {
                  "name": "steak",
                  "quantity": "3",
                },
              ],
              "name": "steak",
            },
          ],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <input
                    onChange={[Function]}
                    placeholder="Recipe Search"
                    type="text"
          />
          <List
                    recipes={
                              Array [
                                        Object {
                                          "id": "1",
                                          "ingredients": Array [
                                            Object {
                                              "name": "steak",
                                              "quantity": "3",
                                            },
                                          ],
                                          "name": "steak",
                                        },
                                      ]
                    }
          />
</div>,
        "_debugID": 4,
        "_renderedOutput": <div>
          <input
                    onChange={[Function]}
                    placeholder="Recipe Search"
                    type="text"
          />
          <List
                    recipes={
                              Array [
                                        Object {
                                          "id": "1",
                                          "ingredients": Array [
                                            Object {
                                              "name": "steak",
                                              "quantity": "3",
                                            },
                                          ],
                                          "name": "steak",
                                        },
                                      ]
                    }
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RecipeSearch
    recipes={
        Array [
            Object {
              "id": "1",
              "ingredients": Array [
                Object {
                  "name": "steak",
                  "quantity": "3",
                },
              ],
              "name": "steak",
            },
            Object {
              "id": "2",
              "ingredients": Array [
                Object {
                  "name": "chicken",
                  "quantity": "2",
                },
              ],
              "name": "chicken",
            },
          ]
    }
/>,
}
`;

exports[`RecipeSearch renders List 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <input
        onChange={[Function]}
        placeholder="Recipe Search"
        type="text"
    />
    <List
        recipes={
            Array [
                Object {
                  "id": "1",
                  "ingredients": Array [
                    Object {
                      "name": "steak",
                      "quantity": "3",
                    },
                  ],
                  "name": "steak",
                },
                Object {
                  "id": "2",
                  "ingredients": Array [
                    Object {
                      "name": "chicken",
                      "quantity": "2",
                    },
                  ],
                  "name": "chicken",
                },
              ]
        }
    />
</div>,
  "nodes": Array [
    <div>
      <input
            onChange={[Function]}
            placeholder="Recipe Search"
            type="text"
      />
      <List
            recipes={
                  Array [
                        Object {
                          "id": "1",
                          "ingredients": Array [
                            Object {
                              "name": "steak",
                              "quantity": "3",
                            },
                          ],
                          "name": "steak",
                        },
                        Object {
                          "id": "2",
                          "ingredients": Array [
                            Object {
                              "name": "chicken",
                              "quantity": "2",
                            },
                          ],
                          "name": "chicken",
                        },
                      ]
            }
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <RecipeSearch
        recipes={
                Array [
                        Object {
                          "id": "1",
                          "ingredients": Array [
                            Object {
                              "name": "steak",
                              "quantity": "3",
                            },
                          ],
                          "name": "steak",
                        },
                        Object {
                          "id": "2",
                          "ingredients": Array [
                            Object {
                              "name": "chicken",
                              "quantity": "2",
                            },
                          ],
                          "name": "chicken",
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": RecipeSearch {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "recipes": Array [
            Object {
              "id": "1",
              "ingredients": Array [
                Object {
                  "name": "steak",
                  "quantity": "3",
                },
              ],
              "name": "steak",
            },
            Object {
              "id": "2",
              "ingredients": Array [
                Object {
                  "name": "chicken",
                  "quantity": "2",
                },
              ],
              "name": "chicken",
            },
          ],
        },
        "refs": Object {},
        "searchRecipes": [Function],
        "state": Object {
          "recipeResults": Array [
            Object {
              "id": "1",
              "ingredients": Array [
                Object {
                  "name": "steak",
                  "quantity": "3",
                },
              ],
              "name": "steak",
            },
            Object {
              "id": "2",
              "ingredients": Array [
                Object {
                  "name": "chicken",
                  "quantity": "2",
                },
              ],
              "name": "chicken",
            },
          ],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <input
                    onChange={[Function]}
                    placeholder="Recipe Search"
                    type="text"
          />
          <List
                    recipes={
                              Array [
                                        Object {
                                          "id": "1",
                                          "ingredients": Array [
                                            Object {
                                              "name": "steak",
                                              "quantity": "3",
                                            },
                                          ],
                                          "name": "steak",
                                        },
                                        Object {
                                          "id": "2",
                                          "ingredients": Array [
                                            Object {
                                              "name": "chicken",
                                              "quantity": "2",
                                            },
                                          ],
                                          "name": "chicken",
                                        },
                                      ]
                    }
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <input
                    onChange={[Function]}
                    placeholder="Recipe Search"
                    type="text"
          />
          <List
                    recipes={
                              Array [
                                        Object {
                                          "id": "1",
                                          "ingredients": Array [
                                            Object {
                                              "name": "steak",
                                              "quantity": "3",
                                            },
                                          ],
                                          "name": "steak",
                                        },
                                        Object {
                                          "id": "2",
                                          "ingredients": Array [
                                            Object {
                                              "name": "chicken",
                                              "quantity": "2",
                                            },
                                          ],
                                          "name": "chicken",
                                        },
                                      ]
                    }
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RecipeSearch
    recipes={
        Array [
            Object {
              "id": "1",
              "ingredients": Array [
                Object {
                  "name": "steak",
                  "quantity": "3",
                },
              ],
              "name": "steak",
            },
            Object {
              "id": "2",
              "ingredients": Array [
                Object {
                  "name": "chicken",
                  "quantity": "2",
                },
              ],
              "name": "chicken",
            },
          ]
    }
/>,
}
`;

exports[`RecipeSearch renders error message if search had no matches 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <input
        onChange={[Function]}
        placeholder="Recipe Search"
        type="text"
    />
    <div>
        Sorry, nothing matched your filter term
    </div>
</div>,
  "nodes": Array [
    <div>
      <input
            onChange={[Function]}
            placeholder="Recipe Search"
            type="text"
      />
      <div>
            Sorry, nothing matched your filter term
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <RecipeSearch
        recipes={
                Array [
                        Object {
                          "id": "1",
                          "ingredients": Array [
                            Object {
                              "name": "steak",
                              "quantity": "3",
                            },
                          ],
                          "name": "steak",
                        },
                        Object {
                          "id": "2",
                          "ingredients": Array [
                            Object {
                              "name": "chicken",
                              "quantity": "2",
                            },
                          ],
                          "name": "chicken",
                        },
                      ]
        }
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": RecipeSearch {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "recipes": Array [
            Object {
              "id": "1",
              "ingredients": Array [
                Object {
                  "name": "steak",
                  "quantity": "3",
                },
              ],
              "name": "steak",
            },
            Object {
              "id": "2",
              "ingredients": Array [
                Object {
                  "name": "chicken",
                  "quantity": "2",
                },
              ],
              "name": "chicken",
            },
          ],
        },
        "refs": Object {},
        "searchRecipes": [Function],
        "state": Object {
          "recipeResults": Array [],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <input
                    onChange={[Function]}
                    placeholder="Recipe Search"
                    type="text"
          />
          <div>
                    Sorry, nothing matched your filter term
          </div>
</div>,
        "_debugID": 6,
        "_renderedOutput": <div>
          <input
                    onChange={[Function]}
                    placeholder="Recipe Search"
                    type="text"
          />
          <div>
                    Sorry, nothing matched your filter term
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RecipeSearch
    recipes={
        Array [
            Object {
              "id": "1",
              "ingredients": Array [
                Object {
                  "name": "steak",
                  "quantity": "3",
                },
              ],
              "name": "steak",
            },
            Object {
              "id": "2",
              "ingredients": Array [
                Object {
                  "name": "chicken",
                  "quantity": "2",
                },
              ],
              "name": "chicken",
            },
          ]
    }
/>,
}
`;
